src/client/content_cao.cpp

```diff
@@ -1657,5 +1658,11 @@ std::string GenericCAO::debugInfoText()
 	return os.str();
 }
 
+bool GenericCAO::doShowSelectionBox()
+{
+	return m_prop.show_selection_box
+		|| (m_client && m_client->getGameUI()->getFlags().show_debug);
```

Comment:
this kinda breaks the abstraction
I suggest moving the show_debug check to `Game::updatePointedThing`